EpikCord/client/http_client.py

```diff
@@ -1,312 +1,233 @@
+            The internal hash of the bucket.
+        external_hash: str
+            The external hash of the bucket.
+        major_parameters: Dict[str, Any]
+            The major parameters for this Bucket
+
+        Attributes
+        ----------
+        internal_hash: str
+            The internal hash of the bucket.
+        external_hash: str
+            The external hash of the bucket.
+        major_parameters: Dict[str, Any]
+            The major parameters for this Bucket
+        """
+        super().__init__(internal_hash=internal_hash, external_hash=external_hash)
+        self.major_parameters: Dict[str, Any] = major_parameters
 
-        return DiscordWSMessage(
-            data=message, type=ws_message.type, extra=ws_message.extra
-        )
+    def __eq__(self, other: TopLevelBucket):
+        if isinstance(other, TopLevelBucket):
+            return self.major_parameters == other.major_parameters
+        return False
 
 
 class HTTPClient:
-    def __init__(self, token: Optional[str] = None, *args, **kwargs):
-        from EpikCord import __version__
-
-        self.base_uri: str = kwargs.pop(
-            "discord_endpoint", "https://discord.com/api/v10"
-        )
-
-        headers = {
-            "User-Agent": f"DiscordBot (https://github.com/EpikCord/EpikCord.py {__version__})",
-            "Content-Type": "application/json",
-        }
-
-        if token:
-            headers["Authorization"] = f"Bot {token}"
-
-        self.session = ClientSession(
-            *args,
-            **kwargs,
-            json_serialize=lambda x, *__, **___: json.dumps(x).decode("utf-8")  # type: ignore
-            if _ORJSON
-            else json.dumps(x),
-            ws_response_class=GatewayWebsocket,
-            headers=headers,
+    def __init__(self, token: str, *, version: int = 10):
+        self.token: str = token
+        self.version: int = version
+        self.session: aiohttp.ClientSession = aiohttp.ClientSession(
+            headers={"Authorization": f"Bot {self.token}"}
         )
-
-        self.ws_connect = self.session.ws_connect
-
-        self.global_ratelimit: asyncio.Event = asyncio.Event()
-        self.global_ratelimit.set()
         self.buckets: Dict[str, Bucket] = {}
+        self.global_event: asyncio.Event = asyncio.Event()
+        self.global_event.set()
+
+    async def ws_connect(self, url: str) -> aiohttp.ClientWebSocketResponse:
+        return await self.session.ws_connect(url)
 
     async def request(
         self,
-        method,
-        url,
+        method: str,
+        url: str,
         *args,
-        attempt: int = 1,
-        to_discord=True,
-        guild_id: Union[str, int] = 0,
-        channel_id: Union[int, str] = 0,
-        reason: Optional[str] = None,
+        discord: bool = True,
+        channel_id: Optional[int] = None,
+        guild_id: Optional[int] = None,
+        webhook_id: Optional[int] = None,
+        webhook_token: Optional[str] = None,
```

Comment:
Can't you just have a dictionary containing all of the parameters that should be filled in for the url? Alternatively, a class that represents the route the request is going to like a lot of other Discord API wrappers do.